ruby/lib/minitest/queue/runner.rb

```diff
@@ -402,6 +413,32 @@ def parser
             queue_config.max_consecutive_failures = Integer(max)
           end
 
+          help = <<~EOS
+            Set the time limit of the execution time from grinds on a given test.
+            For example, when max-test-duration is set to 10 and
+            max-test-duration-percentile is set to 0.5, the test's median execution time during a grind must be
+            lower than 10 milliseconds.
+            The unit is milliseconds and decimal is allowed.
+            Defaults to disabled.
+          EOS
+          opts.on('--max-test-duration LIMIT_IN_MILLISECONDS', *help) do |limit|
+            queue_config.max_test_duration = Float(limit)
+          end
+
+          help = <<~EOS
+            The percentile for max-test-duration. For example, when max-test-duration is set to 10 and
+            max-test-duration-percentile is set to 0.5, the test's median execution time during a grind must be
+            lower than 10 milliseconds.
+            The percentile must be within the range 0 < percentile <= 1.
+            Defaults to 0.5 (50th percentile).
+          EOS
+          opts.on('--max-test-duration-percentile LIMIT_IN_MILLISECONDS', *help) do |percentile|
+            queue_config.max_test_duration_percentile = Float(percentile)
```

Comment:
why not `to_f`?

Reply:
I forgot about .to_f :P I will update it.